tests/store/artifact/test_presigned_url_artifact_repo.py

```diff
@@ -0,0 +1,211 @@
+    CreateDownloadUrlResponse, CreateDownloadUrlRequest, CreateUploadUrlResponse, CreateUploadUrlRequest
+from mlflow.store.artifact.presigned_url_artifact_repo import PresignedUrlArtifactRepository, DIRECTORIES_ENDPOINT
+from mlflow.utils.proto_json_utils import message_to_json
+
+MODEL_NAME = "catalog.schema.model"
+MODEL_VERSION = 1
+MODEL_URI = "/Models/catalog/schema/model/1"
+PRESIGNED_URL_ARTIFACT_REPOSITORY = "mlflow.store.artifact.presigned_url_artifact_repo"
+
+
+def test_artifact_uri():
+    artifact_repo = PresignedUrlArtifactRepository(MODEL_NAME, MODEL_VERSION)
+    assert MODEL_URI == artifact_repo.artifact_uri
+
+
+def mock_list_directory(*args, **kwargs):
+    endpoint = kwargs["endpoint"]
+    json_body = kwargs["json_body"]
+
+    if (endpoint == f'{DIRECTORIES_ENDPOINT}{MODEL_URI}/dir'
+            and json_body == json.dumps({"page_token": "some_token"})):
+        return ListDirectoryResponse(
+            contents=[
+                DirectoryEntry(is_directory=False, path=f"{MODEL_URI}/dir/file2", file_size=2)
+            ]
+        )
+    elif endpoint == f'{DIRECTORIES_ENDPOINT}{MODEL_URI}/dir':
+        return ListDirectoryResponse(
+            contents=[DirectoryEntry(is_directory=False, path=f"{MODEL_URI}/dir/file1", file_size=1)],
+            next_page_token="some_token"
+        )
+    elif endpoint == f'{DIRECTORIES_ENDPOINT}{MODEL_URI}/':
+        return ListDirectoryResponse(
+            contents=[DirectoryEntry(is_directory=True, path=f"{MODEL_URI}/dir")],
+        )
+    else:
+        raise ValueError(f"Unexpected endpoint: {endpoint}")
+
+
+def test_list_artifact_pagination():
+    artifact_repo = PresignedUrlArtifactRepository(MODEL_NAME, MODEL_VERSION)
+
+    with mock.patch(f'{PRESIGNED_URL_ARTIFACT_REPOSITORY}.call_endpoint', side_effect=mock_list_directory) as mock_list:
+        resp = artifact_repo.list_artifacts()
+        assert len(resp) == 1
+        assert resp[0].path == "dir"
+        assert resp[0].is_dir is True
+        assert resp[0].file_size is None
+
+        resp = artifact_repo.list_artifacts("dir")
+        assert len(resp) == 2
+        assert {r.path for r in resp} == {"dir/file1", "dir/file2"}
+        assert {r.is_dir for r in resp} == {False}
+        assert {r.file_size for r in resp} == {1, 2}
+
+        assert mock_list.call_count == 3
+
+
+def _make_pesigned_url(remote_path):
+    return f"presigned_url/{remote_path}"
+
+
+def _make_headers(remote_path):
+    return {f"path_header-{remote_path}": f"remote-path={remote_path}"}
+
+
+def mock_create_download_url(*args, **kwargs):
+    remote_path = json.loads(kwargs["json_body"])["path"]
+    return CreateDownloadUrlResponse(url=_make_pesigned_url(remote_path),
+                                     headers=[HttpHeader(name=header, value=val) for header, val in
+                                              _make_headers(remote_path).items()])
+
```

Comment:
Are there failure cases we should test here as well?

Reply:
I've added failure test cases for all endpoint calling methods. These errors should be propagated to the client.